### DISCUSSION TITLE
Getting error when importing csv file

### DISCUSSION THREAD

**Discussion by adiartawibawa** [filamentphp/filament#13156] [Help]
### Package

Form builder

### Package Version

v3.1

### How can we help you?

Hello everyone,

I'm encountering an issue while trying to import a CSV file using Filament version 3.1. I suspect the problem lies with the default `file-upload.js` script provided by Filament. Here are the details:

1. I'm using Filament v3.1 for my project.
2. When attempting to import a CSV file, an error occurs that seems to originate from `file-upload.js`.

Here are the steps I've already taken:
- Ensured that I am using the latest version of Filament.
- Double-checked my configuration and the script I am using.

Despite these efforts, the issue persists. I have created a video showing the error that occurs when trying to import the CSV file.

Has anyone else experienced a similar issue or have any suggestions on how to resolve this problem?

Thank you in advance for your help!

https://github.com/filamentphp/filament/assets/75978314/4925d7df-78f0-4287-bc97-ed0d5ae45543

**Comment by leandrocfe** [comment]
Could you share the code that you are using?

**Reply by adiartawibawa** [reply]
of course!
The last time was about 3 months ago when I used this import function and everything worked normally. Now that I want to try to import data, it turns out that I encounter problems like in the video that I have included.
This is my code:

ProvinsiResource

```
namespace App\Filament\Resources\ProvinsiResource\Pages;

use App\Filament\Imports\ProvinsiImporter;
use App\Filament\Resources\ProvinsiResource;
use Filament\Actions;
use Filament\Resources\Pages\ManageRecords;

class ManageProvinsis extends ManageRecords
{
    protected static string $resource = ProvinsiResource::class;

    protected function getHeaderActions(): array
    {
        return [
            Actions\CreateAction::make()
                ->icon('heroicon-o-plus-circle'),
            Actions\ImportAction::make()
                ->label('Import')
                ->icon('heroicon-o-arrow-down-tray')
                ->importer(ProvinsiImporter::class),
        ];
    }
}
```
The code of ProvinsiImporter
```
namespace App\Filament\Imports;

use App\Models\Provinsi;
use Filament\Actions\Imports\ImportColumn;
use Filament\Actions\Imports\Importer;
use Filament\Actions\Imports\Models\Import;

class ProvinsiImporter extends Importer
{
    protected static ?string $model = Provinsi::class;

    public static function getColumns(): array
    {
        return [
            ImportColumn::make('code')
                ->requiredMapping()
                ->rules(['required', 'max:2']),
            ImportColumn::make('name')
                ->requiredMapping()
                ->rules(['required', 'max:255']),
        ];
    }

    public function resolveRecord(): ?Provinsi
    {
        return Provinsi::firstOrNew([
            'code' => $this->data['code'],
            'name' => $this->data['name'],
            'meta' => json_encode(['lat' => $this->data['lat'], 'lon' => $this->data['lon']]),
        ]);
    }

    public static function getCompletedNotificationBody(Import $import): string
    {
        $body = 'Your provinsi import has completed and ' . number_format($import->successful_rows) . ' ' . str('row')->plural($import->successful_rows) . ' imported.';

        if ($failedRowsCount = $import->getFailedRowsCount()) {
            $body .= ' ' . number_format($failedRowsCount) . ' ' . str('row')->plural($failedRowsCount) . ' failed to import.';
        }

        return $body;
    }
}
```

**Comment by TwilightDuck** [comment]
Hey just chiming in. I had the same issue. Turns out I had modified the config/livewire.php file.

`  'rules' => 'file|mimes:png,jpg,pdf,csv,mp4,svg,mov,webp|max:1024000', `

When I removed the mimes rule it started working. Not sure why it's not recognizing the csv mime here.

**Reply by sammyaxe** [reply]
This helped, just I think deleting mime rule is bad solution, I would suggest adding csv to the list.
This really should be documented though
